client/signup/steps/domains/index.jsx

```diff
@@ -85,27 +85,13 @@ class DomainsStep extends React.Component {
 	constructor( props ) {
 		super( props );
 
-		const { flowName, signupDependencies } = props;
-		const importSiteUrl = get( signupDependencies, 'importSiteUrl' );
-
-		if ( ( flowName === 'import' || flowName === 'import-onboarding' ) && importSiteUrl ) {
-			this.skipRender = true;
-
-			props.submitSignupStep( {
-				flowName,
-				siteUrl: importSiteUrl,
-				stepName: props.stepName,
-				stepSectionName: props.stepSectionName,
-			} );
-			props.goToNextStep();
-			return;
-		}
-
 		const domain = get( props, 'queryObject.new', false );
 		const search = get( props, 'queryObject.search', false ) === 'yes';
+		const suggestedDomain = get( props, 'signupDependencies.suggestedDomain' );
```

Comment:
Doing things in the constructor like this is very brittle. Please test all the cases when landing on the domain screen with the `suggestedDomain` set like refreshing the browser, going back and forth through the steps.